src/Datadog.Trace/AsyncLocalScopeManager.cs

```diff
@@ -10,51 +9,65 @@ internal class AsyncLocalScopeManager : IScopeManager
 
         private readonly AsyncLocalCompat<Scope> _activeScope = new AsyncLocalCompat<Scope>();
 
-        public event EventHandler<ScopeEventArgs> ScopeOpened;
+        public event EventHandler<SpanEventArgs> SpanOpened;
 
-        public event EventHandler<ScopeEventArgs> ScopeActivated;
+        public event EventHandler<SpanEventArgs> SpanActivated;
 
-        public event EventHandler<ScopeEventArgs> ScopeDeactivated;
+        public event EventHandler<SpanEventArgs> SpanDeactivated;
 
-        public event EventHandler<ScopeEventArgs> ScopeClosed;
+        public event EventHandler<SpanEventArgs> SpanClosed;
+
+        public event EventHandler<SpanEventArgs> TraceEnded;
 
         public Scope Active => _activeScope.Get();
 
         public Scope Activate(Span span, bool finishOnClose)
         {
-            var activeScope = _activeScope.Get();
-            var scope = new Scope(activeScope, span, this, finishOnClose);
-            var scopeOpenedArgs = new ScopeEventArgs(scope);
+            var newParent = Active;
+            var scope = new Scope(newParent, span, this, finishOnClose);
+            var scopeOpenedArgs = new SpanEventArgs(span);
+
+            SpanOpened?.Invoke(this, scopeOpenedArgs);
 
-            ScopeOpened?.Invoke(this, scopeOpenedArgs);
             _activeScope.Set(scope);
 
-            if (activeScope != null)
+            if (scope.Parent != null)
```

Comment:
Even though they will hold the same value, I think that using `newParent` here instead of `scope.Parent` is more readable, since it's the variable you also use several times above and below.
```suggestion
            if (newParent != null)
```